docs/content/en/blog/ci-cd-gap-fulfill-with-pipecd.md

```diff
@@ -0,0 +1,171 @@
+
+If you use container images as an interface between CI and CD, you can take advantage of the LABEL feature of containers to store information about source code commit hashes, branches, etc. The OCI specification also has some predefined LABELs that you can use without defining additional keys.
+
+```dockerfile
+LABEL org.opencontainers.image.source="https://github.com/pipe-cd/pipecd"
+LABEL org.opencontainers.image.revision="0123456789abcdefg"
+```
+
+However, although this approach is similar to the previous one, it is only suitable for artifacts that are container images, and at the same time, scanning container images to get information about commit hashes and displaying it in the UI incurs additional processing overhead.
+
+At this point, we realized a strength in the EventWatcher design: sending events from CI to CD.
+
+The essence of this problem is that when viewing a deployment, you need to know information in the code repository that is only relevant to the manifest repository. In the approach of sending events between CI and CD, you add the necessary information, such as the commit hash and PR URL, as part of the event. When it is piped to create a commit change to the manifest repository, this information is used to link the deployment with the commit trigger and commit the code change.
+
+We use the [git trailer](https://git-scm.com/docs/git-interpret-trailers) feature to add this information to the manifest change commit. And finally, wrap up it as `--context` flag of the pipectl command.
+
+```bash
+pipectl event register --name=event-test --data=ghcr.io/pipecd/helloworld:v0.49.0 \
+--context=Source-Commit-Hash= 6512b1f3fcba7342c11,Source-URL=https://github.com/pipe-cd/pipecd/commit/6512b1f3fcba7342c11,PipeCD-Official-Docs=https://pipecd.dev/
+```
+
+If you're using GitHub as your manifest repo remote storage, you can see it on GitHub UI like this
+
+![](/images/pipecd-commit-context-github-ui.png)
+
+With this feature released under [PipeCD v0.49.3](https://github.com/pipe-cd/pipecd/releases/tag/v0.49.3), users now can have a link from CD to CI for tracking back which application code change causes this deployment in CD pipeline.
+
+## It's good, but it's not enough
+
+After the release of the EventWatcher update, the community provided positive feedback about resolving the connection issues between CI and CD.
+However, we notice some limitations when using PipeCD in very complex projects with many microservices in the same repository.
+
+Each event sent to PipeCD to trigger a deployment can actually be shared to trigger multiple deployments. This is obvious because container images are shared across many different services. For example, changes related to protos or common shared libraries, such as logging and metrics libraries, lead to multiple deployments at once.
```

Comment:
> This is obvious because container images are shared across many different services

In my understanding, each service doesn't share the container images.

Many deployments are triggered by updating some shared libraries across many different services. 👀